### PR TITLE
Added missing description to the KKP CRDs Specs

### PR DESCRIPTION
**Pull request by archups**
**What this PR does / why we need it**:
Updated CRDs docs to add missing descriptions for Cloud providers under Cluster and Preset Spec.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind doc

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
https://docs.kubermatic.com/kubermatic/main/references/crds/
```

### CODE REVIEW

File: pkg/apis/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -939,20 +939,34 @@ type CloudSpec struct {
...
-	Packet              *PacketCloudSpec              `json:"packet,omitempty"`
-	Hetzner             *HetznerCloudSpec             `json:"hetzner,omitempty"`
-	VSphere             *VSphereCloudSpec             `json:"vsphere,omitempty"`
-	GCP                 *GCPCloudSpec                 `json:"gcp,omitempty"`
-	Kubevirt            *KubevirtCloudSpec            `json:"kubevirt,omitempty"`
-	Alibaba             *AlibabaCloudSpec             `json:"alibaba,omitempty"`
-	Anexia              *AnexiaCloudSpec              `json:"anexia,omitempty"`
-	Nutanix             *NutanixCloudSpec             `json:"nutanix,omitempty"`
+	Fake *FakeCloudSpec `json:"fake,omitempty"`
+	// Digitalocean specifies access data to DigitalOcean.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/cluster.go]: "access data" is not the right description. These fields, not just for digitalocean, do not just contain credentials (in fact, in most cases they will not contain those).

These fields contain something like "cloud provider configuration".

File: pkg/apis/kubermatic/v1/cluster.go [hunk 2]
```diff
@@ -1010,19 +1025,19 @@ type AzureCloudSpec struct {
 	// If set to empty string at cluster creation, a new VNet will be created and this field will be updated to
 	// the generated VNet's name.
 	VNetName string `json:"vnet"`
-	// The name of a subnet in the VNet referenced by `vnet`.
+	// SubnetName is the name of a subnet in the VNet referenced by `vnet`.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/cluster.go]: I see what you're doing here, following Go's "Comment should start with the field name" mantra, but IIRC this is not what we want for our CRDs, as it makes the generated documentation just look weird.

File: pkg/apis/kubermatic/v1/cluster.go (hunk 1 above)
Comment (archups) [reply pkg/apis/kubermatic/v1/cluster.go]: I tried to obtain and match here with the actual description available for each cloud provider spec, similar to what I see is done in most of the CRDs. For example [DigitialOceanCloudSpec](https://docs.kubermatic.com/kubermatic/main/references/crds/#digitaloceancloudspec). 
But I am with you, we can make it better as per suggestion. Will fix it for all, thanks!

File: pkg/apis/kubermatic/v1/cluster.go (hunk 2 above)
Comment (archups) [reply pkg/apis/kubermatic/v1/cluster.go]: Hey @xrstf , I got it what are you saying here so should I revert to what was there?

Comment (archups) [reply pkg/apis/kubermatic/v1/cluster.go]: Please suggest!

File: pkg/apis/kubermatic/v1/cluster.go [hunk 3]
```diff
@@ -940,33 +940,33 @@ type CloudSpec struct {
 	ProviderName string `json:"providerName"`
 
 	Fake *FakeCloudSpec `json:"fake,omitempty"`
-	// Digitalocean specifies access data to DigitalOcean.
+	// Digitalocean defines thes configuration data of DigitalOcean Cloud provider.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/cluster.go]: `thes` should be `the`. Also `of DigitalOcean Cloud provider` should probably be `of the DigitalOcean cloud provider`

Comment (archups) [reply pkg/apis/kubermatic/v1/cluster.go]: taken care for all!

File: pkg/apis/kubermatic/v1/cluster.go [hunk 4]
```diff
@@ -956,20 +956,34 @@ type CloudSpec struct {
...
-	Openstack           *OpenstackCloudSpec           `json:"openstack,omitempty"`
-	Packet              *PacketCloudSpec              `json:"packet,omitempty"`
-	Hetzner             *HetznerCloudSpec             `json:"hetzner,omitempty"`
-	VSphere             *VSphereCloudSpec             `json:"vsphere,omitempty"`
-	GCP                 *GCPCloudSpec                 `json:"gcp,omitempty"`
-	Kubevirt            *KubevirtCloudSpec            `json:"kubevirt,omitempty"`
-	Alibaba             *AlibabaCloudSpec             `json:"alibaba,omitempty"`
-	Anexia              *AnexiaCloudSpec              `json:"anexia,omitempty"`
-	Nutanix             *NutanixCloudSpec             `json:"nutanix,omitempty"`
+	Fake *FakeCloudSpec `json:"fake,omitempty"`
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/cluster.go:961]: You could add 

> Fake is a dummy cloud provider that is only used for testing purposes. Do not try to actually use it.

File: pkg/apis/kubermatic/v1/cluster.go [hunk 5]
```diff
@@ -1054,7 +1069,7 @@ type AzureCloudSpec struct {
 	// at cluster creation and `AssignAvailabilitySet` is set to `true`, a new availability set will be created and this field
 	// will be updated to the generated availability set's name.
 	AvailabilitySet string `json:"availabilitySet"`
-
+	// LoadBalancerSKU sets the LB type that will be used for the Azure cluster, possible values are "basic" and "standard", if empty, "basic" will be used
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/cluster.go]: Is missing the final `.`

File: pkg/apis/kubermatic/v1/cluster.go [hunk 6]
```diff
@@ -1184,9 +1199,18 @@ type BringYourOwnCloudSpec struct{}
...
-	AccessKeyID          string `json:"accessKeyID,omitempty"`
-	SecretAccessKey      string `json:"secretAccessKey,omitempty"`
-	AssumeRoleARN        string `json:"assumeRoleARN,omitempty"` //nolint:tagliatelle
+	// The Access key ID used to authenticate against AWS.
+	AccessKeyID string `json:"accessKeyID,omitempty"`
+	// The Secret Access Key used to authenticate against AWS.
+	SecretAccessKey string `json:"secretAccessKey,omitempty"`
+	// Defines the ARN for an IAM role that should be assumed when handling resources on AWS. It will be used
+	// to acquire temporary security credentials using an STS AssumeRole API operation whenever creating an AWS session.
+	// required: false
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/cluster.go]: Is this `required: false` meant to be a comment for the CRD builder? Cause it ain't :grin: The field being `omitempty` already makes it optional, but if you want, you could add `// +optional` to truly signify that.

File: pkg/apis/kubermatic/v1/cluster.go [hunk 7]
```diff
@@ -1337,44 +1361,47 @@ type PreAllocatedDataVolume struct {
...
 
+	// Token is used to authenticate with the Anexia API.
 	Token string `json:"token,omitempty"`
 }
 
 // NutanixCSIConfig contains credentials and the endpoint for the Nutanix Prism Element to which the CSI driver connects.
 type NutanixCSIConfig struct {
 
-	// Prism Element Username for csi driver
+	// Prism Element Username for csi driver.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/cluster.go]: You could capitalize `CSI`

File: pkg/apis/kubermatic/v1/datacenter.go [hunk 8]
```diff
@@ -569,9 +572,12 @@ type DatacenterSpecDigitalocean struct {
 
 // DatacenterSpecOpenstack describes an OpenStack datacenter.
 type DatacenterSpecOpenstack struct {
-	AuthURL          string `json:"authURL"`
+	// Authentication URL
+	AuthURL string `json:"authURL"`
+	// Used to make resources highly available
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/datacenter.go]: Is it though? You have to select 1 AZ to put every VM into. Having multiple AZs might make your Openstack setup more reliable, but here for a datacenter config this is not "making clusters highly available".

File: pkg/apis/kubermatic/v1/etcd_backup_config.go [hunk 9]
```diff
@@ -46,11 +46,12 @@ const (
 // +kubebuilder:subresource:status
 // +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type="date"
 
-// EtcdBackupConfig specifies a add-on.
+// EtcdBackupConfig specifies an add-on.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/etcd_backup_config.go]: This was always a terribly misleading doc string. KKP already has a concept of "addons" and etcd backups have _nothing_ to do with that. A better description would be something along the lines of

> EtcdBackupConfig describes how snapshots of user cluster etcds should be performed. Each user cluster automatically gets a default EtcdBackupConfig in its cluster namespace.

File: pkg/apis/kubermatic/v1/etcd_backup_config.go [hunk 10]
```diff
@@ -83,6 +84,7 @@ type EtcdBackupConfigList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
 
+	// Items is the list of the Etcd backup config.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/etcd_backup_config.go]: "Items is a list of EtcdBackupConfig objects."

File: pkg/apis/kubermatic/v1/preset.go [hunk 11]
```diff
@@ -49,22 +49,38 @@ type Preset struct {
...
-	VSphere             *VSphere             `json:"vsphere,omitempty"`
-	AWS                 *AWS                 `json:"aws,omitempty"`
-	Openstack           *Openstack           `json:"openstack,omitempty"`
-	Packet              *Packet              `json:"packet,omitempty"`
-	GCP                 *GCP                 `json:"gcp,omitempty"`
-	Kubevirt            *Kubevirt            `json:"kubevirt,omitempty"`
-	Alibaba             *Alibaba             `json:"alibaba,omitempty"`
-	Anexia              *Anexia              `json:"anexia,omitempty"`
-	Nutanix             *Nutanix             `json:"nutanix,omitempty"`
+	// Access data for Digital Ocean.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/preset.go]: `DigitalOcean` is the correct name, without a space in between.

File: pkg/apis/kubermatic/v1/preset.go [hunk 12]
```diff
@@ -49,22 +49,38 @@ type Preset struct {
...
+	VSphere *VSphere `json:"vsphere,omitempty"`
+	// Access data for Amazon Web Services(AWS) Cloud.
+	AWS *AWS `json:"aws,omitempty"`
+	// Access data for Openstack.
+	Openstack *Openstack `json:"openstack,omitempty"`
+	// Access data for Packet Cloud.
+	Packet *Packet `json:"packet,omitempty"`
+	// Access data for Google Cloud Platform(GCP).
+	GCP *GCP `json:"gcp,omitempty"`
+	// Access data for Kubervirt.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/preset.go]: If we want to be super duper precise, it's `KubeVirt` with a capital V.

File: pkg/apis/kubermatic/v1/preset.go [hunk 13]
```diff
@@ -49,22 +49,38 @@ type Preset struct {
...
+	Digitalocean *Digitalocean `json:"digitalocean,omitempty"`
+	// Access data for Hetzner.
+	Hetzner *Hetzner `json:"hetzner,omitempty"`
+	// Access data for Microsoft Azure Cloud.
+	Azure *Azure `json:"azure,omitempty"`
+	// Access data for VSphere.
+	VSphere *VSphere `json:"vsphere,omitempty"`
+	// Access data for Amazon Web Services(AWS) Cloud.
+	AWS *AWS `json:"aws,omitempty"`
+	// Access data for Openstack.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/preset.go]: If we want to be precise, it's `OpenStack` with a capital S.

File: pkg/apis/kubermatic/v1/preset.go [hunk 14]
```diff
@@ -140,17 +156,38 @@ func (s Hetzner) IsValid() bool {
...
+	// the generated subnet's name. If no VNet is defined at cluster creation, this field should be empty as well.
+	SubnetName string `json:"subnet,omitempty"`
+	// The name of a route table associated with the subnet referenced by `subnet`.
+	// If set to empty string at cluster creation, a new route table will be created and this field will be updated to
+	// the generated route table's name. If no subnet is defined at cluster creation, this field should be empty as well.
+	RouteTableName string `json:"routeTable,omitempty"`
+	// The name of a security group associated with the subnet referenced by `subnet`.
+	// If set to empty string at cluster creation, a new security group will be created and this field will be updated to
+	// the generated security group's name. If no subnet is defined at cluster creation, this field should be empty as well.
+	SecurityGroup string `json:"securityGroup,omitempty"`
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/preset.go:190]: These are lovely descriptions. So many of ours are just one sentence that is usually not much more than the field name itself, but these here offer real benefit to the end user. Nice.

File: pkg/apis/kubermatic/v1/preset.go [hunk 15]
```diff
@@ -287,6 +346,7 @@ func (s Packet) IsValid() bool {
 type GCP struct {
 	ProviderPreset `json:",inline"`
 
+	// ServiceAccount is the Google Service Account (JSON format), encoded with base64.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/preset.go:349]: :heart:

File: pkg/apis/kubermatic/v1/user_project_binding.go [hunk 16]
```diff
@@ -62,10 +63,11 @@ type UserProjectBindingSpec struct {
 // +kubebuilder:object:generate=true
 // +kubebuilder:object:root=true
 
-// UserProjectBindingList is a list of users.
+// UserProjectBindingList is a list of KKP users and projects binding.
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/user_project_binding.go]: `a list of KKP user and project bindings` (bindings is plural, not users/projects).

File: pkg/apis/kubermatic/v1/preset.go (hunk 11 above)
Comment (archups) [reply pkg/apis/kubermatic/v1/preset.go]: My bad. will fix it.!

File: pkg/apis/kubermatic/v1/cluster.go (hunk 6 above)
Comment (archups) [reply pkg/apis/kubermatic/v1/cluster.go]: I think I tried to replicate the comment. Yes, `// +optional` would be better fit. I will fix this.  👍

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**